<?xml version="1.0"?>
<launch>
  <!-- launch file parameters -->
  <arg name="namespace"     default="denso" />
  <arg name="grid_spacing"  default="15.94" />
  <arg name="samples"       default="20" />
  <arg name="pausetime"     default="2.0" />
  <arg name="use_weights"   default="True" />
  <arg name="debug"         default="True" />
  <arg name="maxdist"       default="1.1" />  <!-- meters -->
  <arg name="mindist"       default="0.8" />  <!-- meters -->
  <arg name="maxtilt"       default="10" />   <!-- degrees -->
  
    <!-- Image view -->
  <node name="left_image_view" pkg="image_view" type="image_view">
    <param name="image_transport" type="string" value="theora" />
    <remap from="image" to="/camera/left/image_raw" />
  </node>
  
  
  <!-- Joint trajectory controller -->
  <include file="$(find denso_control)/launch/joint_trajectory_controller.launch">
    <arg name="namespace" value="$(arg namespace)"/>
  </include>
  
  <!-- camera-robot calibration script -->
  <node name="camera_robot_calibration" pkg="axxb_calibration" type="camera_robot_calibration_camonhand.py" output="screen" required="True">
    <param name="robot_name"    type="string" value="$(arg namespace)" />
    <param name="debug"         type="bool"   value="$(arg debug)"/>
    <param name="grid_spacing"  type="double" value="$(arg grid_spacing)" />
    <param name="use_weights"   type="bool"   value="$(arg use_weights)"/>
    <param name="pausetime"     type="double" value="$(arg pausetime)"/>
    <param name="samples"       type="int"    value="$(arg samples)"/>
    <param name="maxtilt"       type="double" value="$(arg maxtilt)"/>
    <!-- Distances of the pattern to the camera -->
    <param name="maxdist"       type="double" value="$(arg maxdist)"/>
    <param name="mindist"       type="double" value="$(arg mindist)"/>
  </node>
</launch>
